provisioning_templates/snippet/preseed_networking_netplan_setup.erb

```diff
@@ -0,0 +1,62 @@
+<%#
+kind: snippet
+name: preseed_networking_netplan_setup
+model: ProvisioningTemplate
+snippet: true
+description: this will configure host networking on Ubuntu 18.04+ using Netplan.io, it configures your primary interface as well
+    as other configures NICs. It supports physical, VLAN and Alias interfaces. It's intended to be
+    called in your preseed finish template.
+%>
+<% host_subnet = @host.subnet -%>
+<% host_dhcp = host_subnet.nil? ? false : host_subnet.dhcp_boot_mode? -%>
+<% host_subnet6 = @host.subnet6 -%>
+<% host_dhcp6 = host_subnet6.nil? ? false : host_subnet6.dhcp_boot_mode? -%>
+
+real=`ip -o link | awk '/<%= @host.mac -%>/ {print $2;}' | sed s/://`
+cat << EOF > /etc/netplan/01-netcfg.yaml
+# This file describes the network interfaces available on your system
+# For more information, see netplan(5).
+network:
+  version: 2
+  ethernets:
+    #<%= @host.primary_interface.identifier %>
+    $real:
+<% if host_dhcp6 -%>
+      dhcp6: true
+<% end -%>
+<% if host_dhcp -%>
+      dhcp4: true
+<% else -%>
+      addresses: [ <%= @host.ip %>/<%= host_subnet.cidr %><% if @host.ip6 && host_subnet6 && !host_dhcp6 -%>, <%= @host.ip6 %>/<%= host_subnet6.cidr %><% end -%> ]
+      <%if @host.ip  && host_subnet  && !host_dhcp  && host_subnet.gateway  -%>gateway4: <%= host_subnet.gateway  %><% end %>
+      <%if @host.ip6 && host_subnet6 && !host_dhcp6 && host_subnet6.gateway -%>gateway6: <%= host_subnet6.gateway %><% end %>
+      nameservers:
+          search: [ <%= @host.domain %> ]
+          addresses: [ <%= host_subnet.dns_primary %><% if host_subnet.dns_secondary -%>, <%= host_subnet.dns_secondary %><% end -%> ]
+<% end -%>
+EOF
+
+<%# Manage additional interfaces %>
+<% @host.managed_interfaces.each do |interface| -%>
+<% interface_subnet = interface.subnet -%>
+<% interface_dhcp = interface_subnet.nil? ? false : interface_subnet.dhcp_boot_mode? -%>
+<% interface_subnet6 = interface.subnet6 -%>
+<% interface_dhcp6 = interface_subnet6.nil? ? false : interface_subnet6.dhcp_boot_mode? -%>
+<% next if !interface.managed? || (interface_subnet.nil? && interface_subnet6.nil?) || interface.primary -%>
+real=`ip -o link | awk '/<%= interface.mac -%>/ {print $2;}' | sed s/:$//`
+
+cat << EOF >> /etc/netplan/01-netcfg.yaml
+    # <%= interface.identifier %>
+    $real:
+<% if interface_dhcp -%>
+      dhcp4: true
+<% else -%> 
+      addresses: [ <%= interface.ip %>/<%= interface_subnet.cidr %><% if interface.ip6 && interface_subnet6 && !interface_dhcp6 -%>, <%= interface.ip6 %>/<%= interface_subnet6.cidr %><% end -%> ]
+      gateway4: <%= interface_subnet.gateway %>
+        nameservers:
+          search: <%= @interface.domain %>
```

Comment:
maybe here should be a check if a domain is set for the interface, because if no domain is set or the org // location can't access it, it throws an error